evalml/pipelines/classification_pipeline.py

```diff
@@ -136,3 +136,18 @@ def _compute_predictions(self, X, y, objectives, time_series=False):
         if any(not o.score_needs_proba for o in objectives):
             y_predicted = self._predict(X, y, pad=True) if time_series else self._predict(X)
         return y_predicted, y_predicted_proba
+
+    def optimize_threshold(self, X, y, y_pred_proba, objective):
```

Comment:
Maybe this should actually go in `binary_classification_pipeline.py` if it's only used for binary problems?

Reply:
I put it in the classification pipeline class since it also should apply for TimeSeriesBinaryClassificationPipelines, which builds on ClassificationPipeline. Adding tests for that!